clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -875,18 +875,24 @@ rewriteToCallOrInvoke(mlir::Operation *op, mlir::ValueRange callOperands,
                       mlir::Block *landingPadBlock = nullptr) {
   llvm::SmallVector<mlir::Type, 8> llvmResults;
   auto cirResults = op->getResultTypes();
+  auto callIf = cast<mlir::cir::CIRCallOpInterface>(op);
 
   if (converter->convertTypes(cirResults, llvmResults).failed())
     return mlir::failure();
 
+  auto cconv = convertCallingConv(callIf.getCallingConv());
+
   if (calleeAttr) { // direct call
-    if (landingPadBlock)
-      rewriter.replaceOpWithNewOp<mlir::LLVM::InvokeOp>(
+    if (landingPadBlock) {
+      auto newOp = rewriter.replaceOpWithNewOp<mlir::LLVM::InvokeOp>(
           op, llvmResults, calleeAttr, callOperands, continueBlock,
           mlir::ValueRange{}, landingPadBlock, mlir::ValueRange{});
-    else
-      rewriter.replaceOpWithNewOp<mlir::LLVM::CallOp>(op, llvmResults,
-                                                      calleeAttr, callOperands);
+      newOp.setCConv(cconv);
```

Comment:
Don't have an idea on how to avoid 4 dups of saving `newOp` and setting CConv. Different from `CIRCallOpInterface`, LLVM uses `CallOpInterface` from MLIR std, which is not aware of calling convention.

It seems we have to do a dynamic dispatch of at least size 2 (invoke + call) after rewriting. I chose to just leave it clear. It might be more appropriate to refactor this part after we have more similar logic like `setCConv`.

Reply:
I wouldn't worry too much about this. Since we don't need the result for anything else (yet), my suggestion would be:
```
rewriter.replaceOpWithNewOp<mlir::LLVM::InvokeOp>(
          op, llvmResults, calleeAttr, callOperands, continueBlock,
          mlir::ValueRange{}, landingPadBlock, mlir::ValueRange{}).setCConv(cconv);
```